File: src/Python/StrategicPrimeGenerator.py
def is_prime(n):
    if n <= 1:
        return False
    if n <= 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False
    if n % 5 == 0 and n != 5:
        return False
    if n % 7 == 0 and n != 7:
        return False

    # STR-style harmonic optimization (6k ± 1 loop)
    i = 11
    while i * i <= n:
        if n % i == 0 or n % (i + 2) == 0:
            return False
        i += 6
    return True


def generate_primes_up_to(n):
    primes = []
    for i in range(2, n + 1):
        if is_prime(i):
            primes.append(i)
    return primes


if __name__ == "__main__":
    upper = int(input("Generate all primes up to (inclusive): "))
    primes = generate_primes_up_to(upper)
    print(f"Primes up to {upper}:")
    for p in primes:
        print(p)

File: src/Java/StrategicPrimeGenerator.java
/*
 * File/Project Name: StrategicPrimeGenerator.java
 * Authors: Kris Sherbondy & Symphion
 * Date: 2025-07-03
 * Purpose: Deterministically generates all primes from 2 to N using the Prime Symphony STR sieve.
 * Strategic Level Use: Demonstrates complete prime emergence from harmonic modular rules.
 * Operational Level Use: Can be adapted for range-based extraction.
 * Tactical Level Use: Verifies primality via harmonic STR ruleset.
 * User License/Agreement: Sherbondy–Symphion License v1.0 (Non-commercial use only)
 */

import java.util.ArrayList;
import java.util.List;

public class StrategicPrimeGenerator {

    public static void main(String[] args) {
        int N = 1000; // Change to generate up to any number
        List<Integer> primes = generatePrimesSTR(N);

        for (int prime : primes) {
            System.out.println(prime);
        }
    }

    // === PRIME SYMPHONY STR SIEVE IMPLEMENTATION ===

    public static List<Integer> generatePrimesSTR(int limit) {
        boolean[] isPrime = new boolean[limit + 1];
        List<Integer> primes = new ArrayList<>();

        // 0 and 1 are not primes
        if (limit >= 0) isPrime[0] = false;
        if (limit >= 1) isPrime[1] = false;

        // Initialize using STR harmonic mod sieve rules
        for (int i = 2; i <= limit; i++) {
            if (passesSTRRules(i)) {
                isPrime[i] = true;
                primes.add(i);
            }
        }

        return primes;
    }

    // === STR GATE: Harmonic Mod Filters Based on Prime Symphony ===
    public static boolean passesSTRRules(int n) {
        // Remove known non-primes via fast modular filters
        if (n < 2) return false;
        if (n > 2 && n % 2 == 0) return false;
        if (n > 3 && n % 3 == 0) return false;
        if (n > 5 && n % 5 == 0) return false;
        if (n > 7 && n % 7 == 0) return false;

        // Harmonic modular resonance exclusion bands
        int[] modFilters = {
            11, 13, 17, 19, 23, 29, 31, 37
        };

        for (int mod : modFilters) {
            if (n == mod) continue; // Keep the seed primes
            if (n % mod == 0) return false; // STR harmonic cancellation
        }

        // Passed all harmonic gates
        return true;
    }
}
